File: frontend/src/types/authSession.ts
export type AuthRole = {
  id: string;
  name: string;
  description: string | null;
  is_system_role: boolean;
  program_id: string | null;
};

export type AuthSession = {
  tenant_id: string;
  tenant_name: string;
  tenant_slug: string;
  user_id: string;
  email: string;
  full_name: string;
  roles: AuthRole[];
};

File: backend/app/schemas/auth_session.py
import uuid

from pydantic import BaseModel


class AuthRoleResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None = None
    is_system_role: bool
    program_id: uuid.UUID | None = None


class AuthSessionResponse(BaseModel):
    tenant_id: uuid.UUID
    tenant_name: str
    tenant_slug: str
    user_id: uuid.UUID
    email: str
    full_name: str
    roles: list[AuthRoleResponse]